src/txmempool.cpp

```diff
@@ -915,6 +915,13 @@ bool CCoinsViewMemPool::GetCoin(const COutPoint &outpoint, Coin &coin) const {
     // conflict with the underlying cache, and it cannot have pruned entries (as it contains full)
     // transactions. First checking the underlying cache risks returning a pruned entry instead.
     CTransactionRef ptx = mempool.get(outpoint.hash);
+    if (!ptx) {
+        // Also try the cached txns
+        auto it = cache_to_add.find(outpoint.hash);
+        if (it != cache_to_add.end()) {
```

Comment:
If you want to use c++17 niceness, you can put the initializer in the if statement:

```suggestion
        if (auto it = cache_to_add.find(outpoint.hash); it != cache_to_add.end()) {
```

oooh so nice!

https://en.cppreference.com/w/cpp/language/if#If_Statements_with_Initializer